qa/acceptance-tests/src/test/java/io/camunda/it/cluster/DynamicNodeIdTest.java

```diff
@@ -0,0 +1,162 @@
+                              "none"))
+                      .withAdditionalProperties(
+                          Map.of(
+                              "camunda.cluster.size",
+                              CLUSTER_SIZE,
+                              "camunda.cluster.dynamic-node-id.type",
+                              "s3",
+                              "camunda.cluster.dynamic-node-id.s3.taskId",
+                              taskId(Integer.parseInt(app.nodeId().id())),
+                              "camunda.cluster.dynamic-node-id.s3.bucketName",
+                              BUCKET_NAME,
+                              "camunda.cluster.dynamic-node-id.s3.leaseDuration",
+                              LEASE_DURATION,
+                              "camunda.cluster.dynamic-node-id.s3.endpoint",
+                              S3.getEndpoint(),
+                              "camunda.cluster.dynamic-node-id.s3.region",
+                              S3.getRegion(),
+                              "camunda.cluster.dynamic-node-id.s3.accessKey",
+                              S3.getAccessKey(),
+                              "camunda.cluster.dynamic-node-id.s3.secretKey",
+                              S3.getSecretKey())))
+          .build();
+
+  @BeforeAll
+  // Setup as static so it's done before the cluster
+  public static void setupAll() {
+    s3Client =
+        S3NodeIdRepository.buildClient(
+            new S3ClientConfig(
+                Optional.of(new S3ClientConfig.Credentials(S3.getAccessKey(), S3.getSecretKey())),
+                Optional.of(Region.of(S3.getRegion())),
+                Optional.of(S3.getEndpoint())));
+
+    // bucket must be created before the application is started
+    s3Client.createBucket(b -> b.bucket(BUCKET_NAME));
+  }
+
+  @Test
+  public void shouldStartAppCorrectlyAndAcquireALease() throws IOException {
+    // then
+    final var objectsInBucket = s3Client.listObjects(b -> b.bucket(BUCKET_NAME)).contents();
+
+    //    Awaitility.await("Repository has been initialized")
+    //        .untilAsserted(() -> assertThat(objectsInBucket).hasSize(clusterSize));
+    assertThat(objectsInBucket).hasSize(CLUSTER_SIZE);
+    final var leases = new ArrayList<Lease>();
+    for (final var object : objectsInBucket) {
+      final var lease = s3Client.getObject(b -> b.bucket(BUCKET_NAME).key(object.key()));
+      final var payload = lease.readAllBytes();
+      if (payload.length > 0) {
+        final var parsed = Lease.fromJsonBytes(OBJECT_MAPPER, payload);
+        if (parsed.isStillValid(System.currentTimeMillis(), LEASE_DURATION)) {
+          leases.add(parsed);
+        }
+      }
+    }
+
+    assertThat(leases).hasSize(3);
+    final var configuredTaskIds =
+        testCluster.brokers().values().stream()
+            .map(
+                app ->
+                    app.property("camunda.cluster.dynamic-node-id.s3.taskId", String.class, null))
```

Comment:
❓ Isn't it easier to set explicit task IDs in the test set up than dealing with nulls? 🤷

Reply:
This is for getting the value of the property, not for setting it.

Reply:
No I get it, I meant the assertions seemed to test that we also have one set there, because then we were asserting that there was something that we got :smile: